libcxx/include/__ranges/view_interface.h

```diff
@@ -51,16 +51,24 @@ class view_interface {
 public:
   template <class _D2 = _Derived>
   [[nodiscard]] _LIBCPP_HIDE_FROM_ABI constexpr bool empty()
```

Comment:
Although unrelated to this pull request, I'm curious about the reason behind this method being a member template. It appears that this differs from the implementation in libstdc++.

Reply:
This *could* be related to a Clang bug when we implemented this early on. I vaguely remember that there were bugs related to constraints on member functions in Clang a while back, and I think we had to work around it in a few places. But it could also be my memory playing tricks on me, so it's worth double-checking that information.

Reply:
> This _could_ be related to a Clang bug when we implemented this early on. I vaguely remember that there were bugs related to constraints on member functions in Clang a while back, and I think we had to work around it in a few places. But it could also be my memory playing tricks on me, so it's worth double-checking that information.

Ah I see. Thanks for the explanation.